src/main/java/de/numcodex/feasibility_gui_backend/query/broker/direct/DirectBrokerClient.java

```diff
@@ -1,229 +1,215 @@
-                    });
-        } catch (Exception e) {
-            throw new IOException("An error occurred while publishing the query with ID: " + brokerQueryId, e);
-        }
+  }
+
+  /**
+   * Updates a query status in all registered listeners.
+   * @param queryId the id of the query to update
+   * @param queryStatus the {@link QueryStatus} to publish to the listeners
+   */
+  protected void updateQueryStatus(String queryId, QueryStatus queryStatus) {
+    var statusUpdate = new QueryStatusUpdate(this, queryId, SITE_1_ID, queryStatus);
+    var associatedBackendQueryId = brokerToBackendQueryIdMapping.get(queryId);
+    listeners.forEach(
+        l -> l.onClientUpdate(associatedBackendQueryId, statusUpdate)
+    );
+  }
+
+  /**
+   * Updates a query status in all registered listeners.
+   * @param query the query to update
+   * @param queryStatus the {@link QueryStatus} to publish to the listeners
+   */
+  protected void updateQueryStatus(DirectQuery query, QueryStatus queryStatus) {
+    updateQueryStatus(query.getQueryId(), queryStatus);
+  }
+
+  /**
+   * A data container representing a query used for direct communications with a Flare instance.
+   */
+  public static class DirectQuery {
+
+    @Getter
+    private final String queryId;
+    private final Map<String, String> queryDefinitions;
+    private final Map<String, Integer> resultsBySite;
+
+    private DirectQuery(String queryId) {
+      this.queryId = queryId;
+      queryDefinitions = new HashMap<>();
+      resultsBySite = new HashMap<>();
     }
 
-    @Override
-    public void closeQuery(String brokerQueryId) throws QueryNotFoundException {
-        if (brokerQueries.remove(brokerQueryId) == null) {
-            throw new QueryNotFoundException(brokerQueryId);
-        }
+    /**
+     * Creates a new {@link DirectQuery} with a random UUID as a query ID.
+     *
+     * @return The created query.
+     */
+    public static DirectQuery create() {
+      return new DirectQuery(UUID.randomUUID().toString());
     }
 
-    @Override
-    public int getResultFeasibility(String brokerQueryId, String siteId) throws QueryNotFoundException, SiteNotFoundException {
-        return findQuery(brokerQueryId).getSiteResult(siteId);
+    /**
+     * Adds a query definition. A query definition is a query in a specific format (i.e. structured
+     * query / CQL). The specified mime type defines the format of the query itself. When invoked
+     * multiple times for a single mime type, any already existing query content associated with
+     * this mime type gets overwritten.
+     * <p>
+     * For more information on available mime types see {@link QueryMediaType}.
+     *
+     * @param mimeType The mime type defining the format of the query.
+     * @param content The actual query in its string representation.
+     */
+    public void addQueryDefinition(String mimeType, String content) {
```

Comment:
Why is a String used here and the enum used in `getQueryDefinition`? If possible, I would use the enum everywhere. There is also en `EnumMap` which can be used to map values by enum keys.